### PR TITLE
Merge ERC7535 integration into VaultAuction

### PR DESCRIPTION
**Pull request by Tranquil-Flow**

### CODE REVIEW

File: src/core/StrategyVaultETH.sol [hunk 1]
```diff
@@ -76,6 +74,7 @@ contract StrategyVaultETH is Initializable, StrategyVaultETHStorage, ERC4626ETHM
      * @param _whitelistedDeposit Whether the deposit function is whitelisted or not.
      * @param _upgradeable Whether the StrategyVault is upgradeable or not.
      * @param _oracle The oracle implementation to use for the vault.
+     * @param _stakerReward The address of the StakerReward contract.
```
Comment (asanson1404) [inline src/core/StrategyVaultETH.sol:77]: Delete this comment

File: src/core/StrategyVaultETH.sol [hunk 2]
```diff
@@ -87,13 +86,19 @@ contract StrategyVaultETH is Initializable, StrategyVaultETHStorage, ERC4626ETHM
        address _oracle
     ) external initializer {
 
+        // Initialize parent contracts (ERC7535MultiRewardVault)
+        __ERC7535MultiRewardVault_init(_oracle);
+
+        // Initialize the contract
+        __StrategyVaultETH_init_unchained(_nftId, _stratVaultCreator, _whitelistedDeposit, _upgradeable, _stakerReward);
```
Comment (asanson1404) [inline src/core/StrategyVaultETH.sol:93]: Delete _stakerReward as an argument of __StrategyVaultETH_init_unchained

File: src/core/StrategyVaultETH.sol [hunk 3]
```diff
@@ -87,13 +86,19 @@ contract StrategyVaultETH is Initializable, StrategyVaultETHStorage, ERC4626ETHM
...
+    }
+
+    function __StrategyVaultETH_init_unchained(uint256 _nftId, address _stratVaultCreator, bool _whitelistedDeposit, bool _upgradeable, address _stakerReward) internal onlyInitializing {
```
Comment (asanson1404) [inline src/core/StrategyVaultETH.sol:96]: delete _stakerReward

File: src/core/StrategyVaultETH.sol [hunk 4]
```diff
@@ -87,13 +86,19 @@ contract StrategyVaultETH is Initializable, StrategyVaultETHStorage, ERC4626ETHM
...
         // Set up the vault state variables
         stratVaultNftId = _nftId;
         whitelistedDeposit = _whitelistedDeposit;
-        upgradeable = _upgradeable;        
-
-        // Initialize the ERC4626ETHMultiRewardVault
-        __ERC4626ETHMultiRewardVault_init(_oracle);
+        upgradeable = _upgradeable;     
+        stakerReward = _stakerReward;
```
Comment (asanson1404) [inline src/core/StrategyVaultETH.sol:101]: delete as it is set in the constructor

File: src/core/StrategyVaultETH.sol [hunk 5]
```diff
@@ -110,7 +115,7 @@ contract StrategyVaultETH is Initializable, StrategyVaultETHStorage, ERC4626ETHM
     receive() external override payable {
         // TODO: emit an event to notify
     }
-
+
```
Comment (asanson1404) [inline src/core/StrategyVaultETH.sol:118]: unnecessary change lol

File: src/core/StrategyVaultETHStorage.sol [hunk 6]
```diff
@@ -47,6 +47,10 @@ abstract contract StrategyVaultETHStorage is IStrategyVaultETH {
 
     /* ============== STATE VARIABLES ============== */
 
+    /// @notice The address of the StakerReward contract
+    // TODO: Make immutable?
+    address public stakerReward;
```
Comment (asanson1404) [inline src/core/StrategyVaultETHStorage.sol:52]: Just have look above hehe

File: src/core/StrategyVaultManager.sol [hunk 7]
```diff
@@ -63,16 +63,18 @@ contract StrategyVaultManager is
      * @param upgradeable If true, the Strategy Vault is upgradeable. If false, the Strategy Vault is not upgradeable.
      * @param operator The address for the operator that this StrategyVault will delegate to.
      * @param oracle The oracle implementation to use for the vault.
+     * @param stakerReward The address of the StakerReward contract.
```
Comment (asanson1404) [inline src/core/StrategyVaultManager.sol:66]: to delete

File: src/core/StrategyVaultManager.sol [hunk 8]
```diff
@@ -63,16 +63,18 @@ contract StrategyVaultManager is
...
      * @return The address of the newly created StrategyVaultETH.
      */
     function createStratVaultETH(
         bool whitelistedDeposit,
         bool upgradeable,
         address operator,
-        address oracle
+        address oracle,
+        address stakerReward
```
Comment (asanson1404) [inline src/core/StrategyVaultManager.sol:74]: to delete

File: src/core/StrategyVaultManager.sol [hunk 9]
```diff
@@ -63,16 +63,18 @@ contract StrategyVaultManager is
...
     ) public returns (address) {
         // Create a Native ETH StrategyVault
-        IStrategyVaultETH newStratVault = _deployStrategyVaultETH(whitelistedDeposit, upgradeable, oracle);
+        IStrategyVaultETH newStratVault = _deployStrategyVaultETH(whitelistedDeposit, upgradeable, oracle, stakerReward);
```
Comment (asanson1404) [inline src/core/StrategyVaultManager.sol:77]: to delete

File: src/core/StrategyVaultManager.sol [hunk 10]
```diff
@@ -96,11 +99,12 @@ contract StrategyVaultManager is
         bool whitelistedDeposit,
         bool upgradeable,
         address operator,
-        address oracle
+        address oracle,
+        address stakerReward
```
Comment (asanson1404) [inline src/core/StrategyVaultManager.sol:103]: saaame

File: src/core/StrategyVaultManager.sol [hunk 11]
```diff
@@ -96,11 +99,12 @@ contract StrategyVaultManager is
...
     ) external payable returns (address) {
 
         // Create a Native ETH StrategyVault
-        IStrategyVaultETH newStratVault = IStrategyVaultETH(createStratVaultETH(whitelistedDeposit, upgradeable, operator, oracle));
+        IStrategyVaultETH newStratVault = IStrategyVaultETH(createStratVaultETH(whitelistedDeposit, upgradeable, operator, oracle, stakerReward));
```
Comment (asanson1404) [inline src/core/StrategyVaultManager.sol:107]: saaaame

File: src/core/StrategyVaultManager.sol [hunk 12]
```diff
@@ -273,9 +277,10 @@ contract StrategyVaultManager is
      * @param whitelistedDeposit If false, anyone can deposit into the Strategy Vault. If true, only whitelisted addresses can deposit into the Strategy Vault.
      * @param upgradeable If true, the Strategy Vault is upgradeable. If false, the Strategy Vault is not upgradeable.
      * @param oracle The oracle implementation to use for the vault.
+     * @param stakerReward The address of the StakerReward contract.
      * @return The address of the newly deployed Strategy Vault.
      */
-    function _deployStrategyVaultETH(bool whitelistedDeposit, bool upgradeable, address oracle) internal returns (IStrategyVaultETH) {
+    function _deployStrategyVaultETH(bool whitelistedDeposit, bool upgradeable, address oracle, address stakerReward) internal returns (IStrategyVaultETH) {
```
Comment (asanson1404) [inline src/core/StrategyVaultManager.sol:283]: deleteeeee

File: src/interfaces/IStrategyVaultETH.sol [hunk 13]
```diff
@@ -108,21 +116,21 @@ interface IStrategyVaultETH is IStrategyVault {
    */
   function createEigenPod() external;
 
-  /* ============== ERRORS ============== */
+    /* ============== ERRORS ============== */
 
-  /// @dev Returned when not privided the right number of nodes 
-  error InvalidClusterSize();
+    /// @dev Returned when not provided the right number of nodes 
+    error InvalidClusterSize();
```
Comment (asanson1404) [inline src/interfaces/IStrategyVaultETH.sol:122]: Oups forgot it but you can remove that error as we don't need it anymore

File: src/interfaces/IStrategyVaultETH.sol [hunk 14]
```diff
@@ -12,27 +14,27 @@ interface IStrategyVaultETH is IStrategyVault {
...
-  function initialize(
-    uint256 _nftId,
-    address _stratVaultCreator,
-    bool _whitelistedDeposit,
-    bool _upgradeable,
-    address _oracle
-  ) external;
-
-  /* ============== EXTERNAL FUNCTIONS ============== */
+    /* ============== INITIALIZER ============== */
```
Comment (asanson1404) [inline src/interfaces/IStrategyVaultETH.sol:17]: unecessary changes

File: src/interfaces/IStrategyVaultETH.sol [hunk 15]
```diff
@@ -2,6 +2,8 @@
 pragma solidity ^0.8.20;
 
 import {BeaconChainProofs} from "eigenlayer-contracts/libraries/BeaconChainProofs.sol";
+import {IDelegationManager} from "eigenlayer-contracts/interfaces/IDelegationManager.sol";
+import {IStrategy} from "eigenlayer-contracts/interfaces/IStrategy.sol";
```
Comment (asanson1404) [inline src/interfaces/IStrategyVaultETH.sol:6]: Don't think we need these import here

File: src/interfaces/IStrategyVaultETH.sol [hunk 16]
```diff
@@ -108,21 +116,21 @@ interface IStrategyVaultETH is IStrategyVault {
...
 
-  /// @dev Returned when trying to deposit an incorrect amount of ETH. Can only deposit a multiple of 32 ETH. (32, 64, 96, 128, etc.)
-  error CanOnlyDepositMultipleOf32ETH();
+    /// @dev Returned when trying to deposit an incorrect amount of ETH. Can only deposit a multiple of 32 ETH. (32, 64, 96, 128, etc.)
+    error CanOnlyDepositMultipleOf32ETH();
 
-  /// @dev Returned when trying to access DV data but no ETH has been deposited
-  error NativeRestakingNotActivated();
+    /// @dev Returned when trying to access DV data but no ETH has been deposited
+    error NativeRestakingNotActivated();
```
Comment (asanson1404) [inline src/interfaces/IStrategyVaultETH.sol:128]: You can remove that error too
Sorry I forgot

File: src/interfaces/IStrategyVaultManager.sol [hunk 17]
```diff
@@ -21,13 +21,15 @@ interface IStrategyVaultManager {
      * @param upgradeable If true, the Strategy Vault is upgradeable. If false, the Strategy Vault is not upgradeable.
      * @param operator The address for the operator that this StrategyVault will delegate to.
      * @param oracle The oracle implementation to use for the vault.
+     * @param stakerReward The address of the StakerReward contract.
```
Comment (asanson1404) [inline src/interfaces/IStrategyVaultManager.sol:24]: To remove

File: src/interfaces/IStrategyVaultManager.sol [hunk 18]
```diff
@@ -21,13 +21,15 @@ interface IStrategyVaultManager {
...
      * @return The address of the newly created StrategyVaultETH.
      */
     function createStratVaultETH(
         bool whitelistedDeposit,
         bool upgradeable,
         address operator,
-        address oracle
+        address oracle,
+        address stakerReward
```
Comment (asanson1404) [inline src/interfaces/IStrategyVaultManager.sol:32]: To remove

File: src/interfaces/IStrategyVaultManager.sol [hunk 19]
```diff
@@ -38,6 +40,7 @@ interface IStrategyVaultManager {
      * @param upgradeable If true, the Strategy Vault is upgradeable. If false, the Strategy Vault is not upgradeable.
      * @param operator The address for the operator that this StrategyVault will delegate to.
      * @param oracle The oracle implementation to use for the vault.
+     * @param stakerReward The address of the StakerReward contract.
```
Comment (asanson1404) [inline src/interfaces/IStrategyVaultManager.sol:43]: Same

File: src/interfaces/IStrategyVaultManager.sol [hunk 20]
```diff
@@ -47,7 +50,8 @@ interface IStrategyVaultManager {
         bool whitelistedDeposit,
         bool upgradeable,
         address operator,
-        address oracle
+        address oracle,
+        address stakerReward
```
Comment (asanson1404) [inline src/interfaces/IStrategyVaultManager.sol:54]: Same

File: test/Auction.t.sol [hunk 21]
```diff
@@ -626,7 +626,7 @@ contract AuctionTest is ByzantineDeployer {
 
     function _createStratVaultETH() internal returns (address) {
         vm.prank(alice);
-        return strategyVaultManager.createStratVaultETH(true, true, ELOperator1, address(0));
+        return strategyVaultManager.createStratVaultETH(true, true, ELOperator1, address(0), address(0));
```
Comment (asanson1404) [inline test/Auction.t.sol:629]: To remove as StakerRewards is set in the constructor of the Beacon Proxy implementation

**Review by asanson1404 — CHANGES_REQUESTED** [review]

File: src/core/StrategyVaultETH.sol (hunk 5 above)
Comment (Tranquil-Flow) [reply src/core/StrategyVaultETH.sol:116]: but otherwise there is an extra line compared to the other sections of code D:

File: src/core/StrategyVaultETHStorage.sol (hunk 6 above)
Comment (asanson1404) [reply src/core/StrategyVaultETHStorage.sol:52]: You forgot that

File: src/interfaces/IStrategyVaultETH.sol (hunk 15 above)
Comment (asanson1404) [reply src/interfaces/IStrategyVaultETH.sol:6]: You forgot that

File: src/interfaces/IStrategyVaultETH.sol (hunk 16 above)
Comment (asanson1404) [reply src/interfaces/IStrategyVaultETH.sol:122]: same you can delete
